### PR TITLE
Compress face and line data

### PR DESCRIPTION
**Pull request by drwells**
I found two places where we can use `std::vector<bool>` to store data as bitfields instead of storing `unsigned char`s or `ReferenceCell`s. This lowers memory use by about 3% (712 MB -> 690 MB) for my big tet mesh.

### CODE REVIEW

File: source/grid/tria.cc [hunk 1]
```diff
@@ -2358,9 +2351,15 @@ namespace internal
...
+                    const unsigned char raw_orientation =
+                      connectivity.entity_orientations(1)[k];
+                    // it doesn't make sense to set any flags except
+                    // orientation for a line
+                    Assert(raw_orientation == 0 || raw_orientation == 1,
+                           ExcInternalError());
                     faces.quads_line_orientations
                       [q * GeometryInfo<2>::faces_per_cell + j] =
-                      connectivity.entity_orientations(1)[k];
+                      bool(raw_orientation);
```
Comment (tjhei) [inline source/grid/tria.cc:2362]: Can you do ==1 instead of casting to bool?

File: include/deal.II/grid/tria_faces.h [hunk 2]
```diff
@@ -70,14 +70,29 @@ namespace internal
        *
        * @note Used only for dim=3.
        */
-      std::vector<unsigned char> quads_line_orientations;
+      std::vector<bool> quads_line_orientations;
```
Comment (tjhei) [inline include/deal.II/grid/tria_faces.h:73]: Can you update the comment to say true means flipped (I assume this is what it is)?

**Review by tjhei — APPROVED** [review]

File: include/deal.II/grid/tria_faces.h (hunk 2 above)
Comment (drwells) [reply include/deal.II/grid/tria_faces.h:73]: Yes. I'm also writing more general documentation for how we handle orientations since its a tricky area.

Comment (drwells) [reply include/deal.II/grid/tria_faces.h:73]: FYI: unlike 'flip' and 'rotate', for 'orientation' `true` is the default and `false` is the reverse.

**Review by tjhei — APPROVED** [review]

**Pull request merged**